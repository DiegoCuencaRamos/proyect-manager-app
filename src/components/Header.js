import React from 'react'
import { Link } from 'react-router-dom'
import { useDispatch } from 'react-redux'
import { startLogout } from '../actions/auth'

const Header = () => {
    const dispatch = useDispatch()
    
    return (
        <header className="header">
            <div className="container">
                <div className="header__content">
                    <Link to={'/dashboard'}>
                        <span className="header__title">Proyect Manager</span>
                    </Link>
                    <button
                        className="button--link"
                        onClick={() => dispatch(startLogout())}
                    >Logout</button>
                </div>
            </div>
        </header>
)}

export default Header