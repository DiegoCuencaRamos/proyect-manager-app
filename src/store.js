import { createStore } from 'redux'
import rootReducer from './reducer'

const enhancer = window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()

const store = createStore(
    rootReducer,
    /* preloadedState. */
    enhancer,
)

export default store