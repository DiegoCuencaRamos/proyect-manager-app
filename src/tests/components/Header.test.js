import React from 'react'
import { mount, render } from 'enzyme'
import { MemoryRouter } from 'react-router'
import { Provider } from 'react-redux'
import store from '../../store'
import Header from '../../components/Header'

const getWrapper = () => mount(
    <MemoryRouter>
        <Provider store={store}>
            <Header />
        </Provider>
    </MemoryRouter>
)

test('Shoud render Header correctly', () => {
    const wrapper = render(
        <MemoryRouter>
            <Provider store={store}>
                <Header />
            </Provider>
        </MemoryRouter>
    )
    expect(wrapper).toMatchSnapshot()
})

// Falta comprobar el onLogOut event