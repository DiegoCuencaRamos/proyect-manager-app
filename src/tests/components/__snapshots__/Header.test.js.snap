// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shoud render Header correctly 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "static",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "static",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "static",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Header>
        <header
          className="header"
        >
          <div
            className="container"
          >
            <div
              className="header__content"
            >
              <Link
                to="/dashboard"
              >
                <LinkAnchor
                  href="/dashboard"
                  navigate={[Function]}
                >
                  <a
                    href="/dashboard"
                    onClick={[Function]}
                  >
                    <span
                      className="header__title"
                    >
                      Proyect Manager
                    </span>
                  </a>
                </LinkAnchor>
              </Link>
              <button
                className="button--link"
                onClick={[Function]}
              >
                Logout
              </button>
            </div>
          </div>
        </header>
      </Header>
    </Provider>
  </Router>
</MemoryRouter>
`;
